Conch/lib/Conch/Controller/DeviceValidation.pm

```diff
@@ -21,11 +22,33 @@ Get latest validation states for a device
 =cut
 
 sub list_validation_states ($c) {
+	my @statuses;
```

Comment:
perhaps this should use the new json schema based validation for api input?

Reply:
🤦‍♂️ 

duh.

Reply:
This is validating the values in query parameters, not in a JSON payload. We can still use JSON::Validator to validate the parameters. Here's what it looks like:

```
	my @statuses;
	if ( $c->param('status') ) {
		@statuses = map { lc($_) } split /,\s*/, $c->param('status');
		my @errors = validate_json(
			\@statuses,
			{
				type  => 'array',
				items => { type => 'string', enum => [ 'pass', 'fail', 'error' ] }
			}
		);
		return $c->status(
			400,
			{
				error =>
					"'status' query parameter must be any of 'pass', 'fail', or 'error'."
			}
		) if @errors;
	}
```

It's a little less obtuse than the previous code, and I like having a standard way of validating any input. Thoughts?